spec/dummy_app/app/models/person.rb

```diff
@@ -54,5 +44,20 @@ def load(value)
     end
   end
 
-  serialize :time_zone, TimeZoneSerializer.new
+  if ActiveRecord.gem_version > Gem::Version.new("5.2")
+    require_relative "../lib/time_zone"
+    attribute :time_zone, ::ActiveModel::Type::TimeZone.new
+  else
+    # Convert strings to TimeZone objects when assigned
+    def time_zone=(value)
+      if value.is_a? ActiveSupport::TimeZone
+        super
+      else
+        zone = ::Time.find_zone(value) # nil if can't find time zone
+        super zone
+      end
+    end
+
+    serialize :time_zone, TimeZoneSerializer.new
+  end
```

Comment:
Does rails 6 change how `serialize` works? If so, can you point me to some reading on the subject?

Reply:
I know the ActiveModel Attribute API was released in 5.1 but I don't know why @iggant made that change. I'll have to dig, or maybe @iggant you can explain

Reply:
Dug into this and the TimeZone Attribute is technically not a requirement to make tests pass on AR 6.0.
Without it, some tests were failing but the reason for the failures weren't directly related do that. I explained the reason in detail in https://github.com/paper-trail-gem/paper_trail/pull/1172/commits/ac12912dcb9f2935938bb48af43979f2a8442b28

That being said, even though adding a new TimeZone Attribute isn't a requirement, it makes the code cleaner. It's not a new 6.0 thing, we could have used it since Rails 5.0 when the Attribute API got released https://api.rubyonrails.org/classes/ActiveRecord/Attributes/ClassMethods.html

Let me know if you prefer to stick with our old serializer.

Reply:
> Dug into this and the TimeZone Attribute is technically not a requirement to make tests pass on AR 6.0.

Thanks for investigating. For now, I have removed this. Let's discuss it in a separate PR.